### ISSUE TITLE
Partner Interconnect attachment name too long

### ISSUE THREAD

**Issue by ianbrown78** [terraform-google-modules/terraform-example-foundation#1127]
### TL;DR

Using the default values within "3-networks-dual-svpc" partner_interconnect.tf in either Australian region, the vlan attachment names violate the max length.
The error provided in Terraform is:
`Error creating InterconnectAttachment: googleapi: Error 400: Invalid value for field 'resource.name': 'vl-onprem-dc2-syd-zone2-1605-d-shared-base-australia-southeast1-cr2'. Must be a match of regex '(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)'`

### Expected behavior

I expect the defaults to produce sane and implementable values across all layers of CFT, and that the plan stage should detect the name value is too long.

### Observed behavior

No error is produced when running terraform plan.
An error is produced when attempting to apply the gcp-networks dual-svpc layer with partner interconnects.
`Error creating InterconnectAttachment: googleapi: Error 400: Invalid value for field 'resource.name': 'vl-onprem-dc2-syd-zone2-1605-d-shared-base-australia-southeast1-cr2'. Must be a match of regex '(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)'`

### Terraform Configuration

```hcl
module "shared_base_interconnect" {
  source = "../partner_interconnect"

  attachment_project_id = local.base_project_id
  vpc_name              = "${var.environment_code}-shared-base"
  preactivate           = true

  region1                         = var.default_region1
  region1_router1_name            = module.base_shared_vpc.region1_router1.router.name
  region1_interconnect1_location  = "syd-zone1-1660"
  region1_interconnect1_onprem_dc = "onprem-dc1"
  region1_router2_name            = module.base_shared_vpc.region1_router2.router.name
  region1_interconnect2_location  = "syd-zone2-1605"
  region1_interconnect2_onprem_dc = "onprem-dc2"

  region2                         = var.default_region2
  region2_router1_name            = module.base_shared_vpc.region2_router1.router.name
  region2_interconnect1_location  = "mel-zone1-1988"
  region2_interconnect1_onprem_dc = "onprem-dc3"
  region2_router2_name            = module.base_shared_vpc.region2_router2.router.name
  region2_interconnect2_location  = "mel-zone2-4843"
  region2_interconnect2_onprem_dc = "onprem-dc4"


  cloud_router_labels = {
    vlan_1 = "cr1",
    vlan_2 = "cr2",
    vlan_3 = "cr3",
    vlan_4 = "cr4"
  }

  depends_on = [
    module.base_shared_vpc
  ]
}
```

### Terraform Version

```sh
Terraform v1.7.4 on linux_amd64
```

### Additional information

I have worked around this issue by creating a `default_region1_short` variable with the first 2 letters of each work plus the region number. (i.e. au-se1)
